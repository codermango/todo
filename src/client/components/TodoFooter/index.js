import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { clearAll } from '../../actions';

import styles from './styles.css';

class TodoFooter extends Component {

  static propTypes = {
    clearAll: PropTypes.func.isRequired
  }

  handleClearAll = () => {
    this.props.clearAll();
  }

  render() {
    return (
      <div className={styles.todoFooter}>
        <button onClick={this.handleClearAll}>Clear All</button>
      </div>
    )
  }
}

const mapDispatchToProps = dispatch => ({
  clearAll: () => dispatch(clearAll())
});

export default connect(null, mapDispatchToProps)(TodoFooter);
