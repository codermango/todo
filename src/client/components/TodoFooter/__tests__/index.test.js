import React from 'react';
import { shallow } from 'enzyme';
import TodoFooter from '../index';


describe('<TodoFooter />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<TodoFooter clearAll={jest.fn()} />);
    expect(wrapper.instance()).toMatchSnapshot();
  });

  it('clearAll is called when clear all button is clicked', () => {
    const wrapper = shallow(<TodoFooter clearAll={jest.fn()} />);
    const instance = wrapper.instance();
    wrapper.find('button').simulate('click');
    expect(instance.props.clearAll).toHaveBeenCalledTimes(1);
  });
});
