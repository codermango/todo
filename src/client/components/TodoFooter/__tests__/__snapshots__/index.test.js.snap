// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoFooter /> renders correctly 1`] = `
TodoFooter {
  "context": Object {},
  "handleClearAll": [Function],
  "props": Object {
    "clearAll": [MockFunction],
  },
  "refs": Object {},
  "setState": [Function],
  "state": null,
  "updater": Updater {
    "_callbacks": Array [],
    "_renderer": ReactShallowRenderer {
      "_context": Object {},
      "_didScheduleRenderPhaseUpdate": false,
      "_dispatcher": Object {
        "readContext": [Function],
        "useCallback": [Function],
        "useContext": [Function],
        "useDebugValue": [Function],
        "useEffect": [Function],
        "useImperativeHandle": [Function],
        "useLayoutEffect": [Function],
        "useMemo": [Function],
        "useReducer": [Function],
        "useRef": [Function],
        "useState": [Function],
      },
      "_element": <TodoFooter
        clearAll={[MockFunction]}
      />,
      "_firstWorkInProgressHook": null,
      "_forcedUpdate": false,
      "_instance": [Circular],
      "_isReRender": false,
      "_newState": null,
      "_numberOfReRenders": 0,
      "_renderPhaseUpdates": null,
      "_rendered": <div
        className="todoFooter"
      >
        <button
          onClick={[Function]}
        >
          Clear All
        </button>
      </div>,
      "_rendering": false,
      "_updater": [Circular],
      "_workInProgressHook": null,
    },
  },
}
`;
